src/BugsnagUnity/Configuration.cs

```diff
@@ -10,17 +10,19 @@ public abstract class AbstractConfiguration : IConfiguration
 
     public const string DefaultSessionEndpoint = "https://sessions.bugsnag.com";
 
-    public AbstractConfiguration(string apiKey)
+    public AbstractConfiguration() {}
+
+    protected virtual void SetupDefaults(string apiKey)
```

Comment:
If all the default values that would be overridden are set here, they won't need to be set in the other configurations.